src/App.jsx

```diff
@@ -14,33 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABETICALLY = 'alphabetically';
+const SORT_BY_LENGTH = 'by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  let visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_ALPHABETICALLY:
+        visibleGoods = visibleGoods.toSorted();
+        if (reversed) {
+          visibleGoods = visibleGoods.toReversed();
+        }
+
+        break;
+      case SORT_BY_LENGTH:
+        if (reversed) {
+          visibleGoods = visibleGoods.toSorted((a, b) => b.length - a.length);
+        } else {
+          visibleGoods = visibleGoods.toSorted((a, b) => a.length - b.length);
+        }
+
+        break;
+      default:
+        return null;
+    }
+  }
```

Comment:
Ideally, we want to have this logic inside a separate function, which is solely responsible for sorting/reversing goods. This is okayish to have it in the component here, treat it as a minor suggestion